kstreams/engine.py

```diff
@@ -155,6 +159,13 @@ def add_stream(self, stream: Stream) -> None:
             stream.deserializer = self.deserializer
         self._streams.append(stream)
 
+    async def remove_stream(self, stream_name: str) -> None:
```

Comment:
I like the way it is now. Just to think about: will you always want to fail when you can't remove?

Reply:
In my use case, i think I prefer that it fails because I would like to be aware that something isn't working as I expect. I also based this decision to throw based just having the python .remove function as an example as a standard.

I can imagine someone asking for this function not to fail, though; I asked the same question to @marcosschroh. He thought my gut to fail was reasonable, so went with that.

Thanks for the comment to think about it; glad you agree that failing is the right call for now, if I understand correctly.

Reply:
good question. `add_stream` expects a `stream instance` as parameter, maybe `remove_stream` should be the same. Then, the end user must call `get_stream` to get the instance and then call `remove_stream`

Reply:
that's a separate question @marcosschroh, but one i'm open to. I think neither makes me totally happy. something about the stream.name felt easier for me as a developer to use, as it's the key for the remove in a way. or were we removing based on the object reference which felt like it had a more specific use case

Reply:
discussed offline. will add changes that remove the exception definition, have remove take in a stream and fail if stream doesn't exist.

Reply:
Yes, I think the notfound is good. If we need to support not raising in the future, I would go for a bool param like `rise_exception = True` in the signature

Reply:
Don't really care either way, but @marcosschroh argued for removing the exception. I leaned toward not having it, since its less code and I don't think adds much.

ValueError: list.remove(x): x not in list exception will be called by the remove if the stream object is not found in the list.

To support not raising, we could add code to the remove function that says: if stream doesn't exist in the stream list, return none before calling the remove

if we wanted the custom exception, I would do a get_stream, check if null, if so, raise custom exception, else, remove stream.

For the bool in the signature, I understand you mean the signature of the remove_stream function